util/client/ngmclient/client.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 PingCAP, Inc. Licensed under Apache-2.0.
+
+package ngmclient
+
+import (
+	"context"
+	"fmt"
+	"net/http/httputil"
+	"net/url"
+	"sync/atomic"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/joomcode/errorx"
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/fx"
+	"golang.org/x/sync/singleflight"
+
+	"github.com/pingcap/tidb-dashboard/pkg/utils/topology"
+	"github.com/pingcap/tidb-dashboard/util/topo"
+)
+
+var (
+	NgmErrNS                 = errorx.NewNamespace("error.api.ng_monitoring")
+	ErrNgMonitoringNotDeploy = NgmErrNS.NewType("not_deploy")
+	ErrNgMonitoringNotStart  = NgmErrNS.NewType("not_start")
```

Comment:
It's wired to mix ngm and ngmonitoring.

```suggestion
	ErrNgmNotDeployed = NgmErrNS.NewType("ngm_not_deployed")
	ErrNgmNotStarted  = NgmErrNS.NewType("ngm_not_started")
```